### PR TITLE
backupccl: update restore/nodeshutdown tests to use new roachtest framework

### PR DESCRIPTION
**Pull request by msbutler**
The restore/nodeshutdown tests have been using a very old workload that will not be restorable when #93804 lands. This patch changes the restore/nodeshutdown workload to a 80GB tpce restore and moves the tests to run on aws instead of gcp.

Release note: None

Epic: None

### CODE REVIEW

File: pkg/cmd/roachtest/tests/restore.go [hunk 1]
```diff
@@ -840,14 +804,16 @@ func (sp *restoreSpecs) run(ctx context.Context, target string) error {
...
 	}
 
-	db, err := sp.c.ConnE(ctx, sp.t.L(), sp.c.Node(1)[0])
+	db, err := sp.c.ConnE(ctx, sp.t.L(), sp.c.Node(node)[0])
 	if err != nil {
-		return 0, errors.Wrap(err, "failed to connect to node 1; running restore detached")
+		return 0, errors.Wrapf(err, "failed to connect to node %d; running restore detached", node)
 	}
 	var jobID jobspb.JobID
 	if err := db.QueryRow(`SELECT job_id FROM [SHOW JOBS] WHERE job_type = 'RESTORE'`).Scan(&jobID); err != nil {
```
Comment (adityamaru) [inline pkg/cmd/roachtest/tests/restore.go:819]: Hmm I'm surprised this works without a retry loop? Don't we see runs where we haven't created the job entry by the time we execute this query?

**Review by adityamaru — APPROVED** [review]

File: pkg/cmd/roachtest/tests/restore.go (hunk 1 above)
Comment (msbutler) [reply pkg/cmd/roachtest/tests/restore.go:819]: looks like a detached restore only returns with the job id once the job record has been written, so I don't think a retry is necessary:
https://github.com/cockroachdb/cockroach/blob/master/pkg/ccl/backupccl/restore_planning.go#L1959

Comment (adityamaru) [reply pkg/cmd/roachtest/tests/restore.go:819]: ah yes sgtm

**Comment by msbutler** [comment]
TFTR!

bors r=adityamaru

**Pull request merged**